llvm/test/Transforms/GVN/PRE/rle.ll

```diff
@@ -98,12 +98,13 @@ define ptr @coerce_mustalias3(float %V, ptr %P) {
 define float @coerce_mustalias4(ptr %P, i1 %cond) {
 ; CHECK-LABEL: @coerce_mustalias4(
 ; CHECK-NEXT:    [[A:%.*]] = load i32, ptr [[P:%.*]], align 4
-; CHECK-NEXT:    [[TMP1:%.*]] = bitcast i32 [[A]] to float
+; CHECK-NEXT:    [[TMP1:%.*]] = load float, ptr [[P]], align 4
 ; CHECK-NEXT:    br i1 [[COND:%.*]], label [[T:%.*]], label [[F:%.*]]
 ; CHECK:       T:
 ; CHECK-NEXT:    ret float [[TMP1]]
 ; CHECK:       F:
-; CHECK-NEXT:    ret float [[TMP1]]
+; CHECK-NEXT:    [[X:%.*]] = bitcast i32 [[A]] to float
+; CHECK-NEXT:    ret float [[X]]
```

Comment:
Why does this fail to fold now? There are no access size changes going on here.

Reply:
sorry, I had messed up which version of opt I had used for UTC